src/Xamarin.Android.Build.Tasks/Tasks/ConvertResourcesCases.cs

```diff
@@ -27,34 +29,44 @@ public class ConvertResourcesCases : Task
 		public string ResourceNameCaseMap { get; set; }
 
 		Dictionary<string,string> resource_name_case_map;
-		Dictionary<string, HashSet<string>> customViewMap;
+		ConcurrentDictionary<string, HashSet<string>> customViewMap;
+		Dictionary<string, string> acw_map = new Dictionary<string, string> ();
 
 		public override bool Execute ()
 		{
 			resource_name_case_map = MonoAndroidHelper.LoadResourceCaseMap (ResourceNameCaseMap);
-			var acw_map = MonoAndroidHelper.LoadAcwMapFile (AcwMapFile);
+			acw_map = MonoAndroidHelper.LoadAcwMapFile (AcwMapFile);
 
 
 			if (CustomViewMapFile != null)
 				customViewMap = Xamarin.Android.Tasks.MonoAndroidHelper.LoadCustomViewMapFile (BuildEngine4, CustomViewMapFile);
 
 			// Look in the resource xml's for capitalized stuff and fix them
-			FixupResources (acw_map);
+			var task = ThreadingTasks.Task.Run (() => {
+				DoExecute ();
+			}, Token);
+
+			task.ContinueWith (Complete).ConfigureAwait (false);
```

Comment:
Side note: I think `ConfigureAwait(false)` will only do something if you `await` it.

Tacking it on the end, I don't think will do anything?